source/reflect.h2

```diff
@@ -1967,13 +1968,14 @@ parse_context: type =
 
     private grab_until_impl: (inout this, in e: std::string, out r: std::string, any: bool) -> bool = 
     {
-        end:= pos;
+        end:= : () -> _ = {
```

Comment:
Is there a ternary operator in cpp2?

Reply:
Not that I know of, this is the only way I could think of at the moment for fixing this lint, of course I could disable it instead, or we could fix the bug because the variable is initialized in both branches in the original `cpp2` code, however I couldn't satisfy `cppfront` by just declaring it, maybe there is a way... and I just don't know it yet.

Reply:
```
        end: size_t;
        if any {
            end = regex..find_first_of(e, pos);
        }
        else {
            end = regex..find(e, pos);
        }
```

this should work in theory, but I get

reflect.h2...
reflect.h2(1973,13): error: local variable r must be initialized before end (local variables must be initialized in the order they are declared)
  ==> program violates initialization safety guarantee - see previous errors

I'm not sure if this is a bug or not, I could move the `r = ""`; at the top instead but that will cause r to be initialized twice if the expression is found which I think is a pessimization.

I consider that it should work no matter the initialization order, or we will force people to write functions with akwardly placed parameters, so IMHO it's a bug! Do you also feel the same, should I open a bug?

Reply:
> reflect.h2(1973,13): error: local variable r must be initialized before end (local variables must be initialized in the order they are declared)

That's as intended, because `r` is an `out` parameter and so for initialization purposes is treated the same as an uninitialized local variable right at the start of the function.

The main intent of requiring uninitialized locals to be initialized in declaration order is to ensure that their lifetimes still correctly nest when they're destroyed in reverse order, so that nested structured lifetime is preserved.

But you're right to point out that doesn't really apply to `out` parameters because those outlive the function anyway, so I could relax this.

Reply:
looking forward to the relaxation